inc/Engine/CriticalPath/CriticalCSSSubscriber.php

```diff
@@ -167,15 +167,19 @@ public function generate_critical_css_on_activation( $old_value, $value ) {
 		}
 
 		try {
-			if ( ( new FilesystemIterator( $critical_css_path, FilesystemIterator::SKIP_DOTS ) )->valid() ) {
-				// Bail out if the folder is not empty.
-				return;
-			}
+			$critical_css_path_file_iterator = ( new FilesystemIterator( $critical_css_path, FilesystemIterator::SKIP_DOTS ) );
 		} catch ( UnexpectedValueException $e ) {
 			// Bail out when folder is invalid.
 			return;
 		}
 
+		// Bail out if this folder has no files (not folders).
+		foreach ( $critical_css_path_file_iterator as $file ) {
+			if ( $file->isFile() ) {
+				return;
+			}
+		}
+
 		// Generate the CPCSS files.
 		$this->critical_css->process_handler();
```

Comment:
I don't think we want to bail out if we find a file. Rather, we want to bail out if there are no directories. 

@Tabrisrp is this right?

if yes, I'd suggest this code instead:

```suggestion
		// Bail out if there are no directories within the critical-css.
		foreach ( $iterator as $entry ) {
			if ( $entry->isDir() ) {
				// Generate the CPCSS files.
				return $this->critical_css->process_handler();;
			}
		}
```

Suggestion 2: Shrink the variable name down to `$iterator` to save memory.